pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -124,6 +124,33 @@ func (dv *openShiftClusterSlimDynamicValidator) Dynamic(ctx context.Context) err
 	return nil
 }
 
+// Validate permissions on a Vnet
+func (dv *openShiftClusterSlimDynamicValidator) ValidateVnetPermissions(ctx context.Context) error {
+	azureEnv, err := azure.EnvironmentFromName(dv.clusterSpec.AZEnvironment)
+	if err != nil {
+		return err
+	}
+	spAuthorizer, err := validateServicePrincipalProfile(ctx, dv.log, dv.spp, dv.tenantID, azureEnv.ResourceManagerEndpoint)
+	if err != nil {
+		return err
+	}
+
+	spPermissions := authorization.NewPermissionsClient(&azureEnv, dv.subscriptionID, spAuthorizer)
+
+	vnetr, err := azure.ParseResourceID(dv.clusterSpec.VNetID)
+	if err != nil {
+		return err
+	}
+
+	err = validateVnetPermissions(ctx, dv.log, spAuthorizer, spPermissions, dv.clusterSpec.VNetID, &vnetr, api.CloudErrorCodeInvalidServicePrincipalPermissions, "provided service principal")
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
It would be simpler to do this:
```suggestion
	return err
```